nifi-registry-docs/src/main/asciidoc/administration-guide.adoc

```diff
@@ -895,3 +895,167 @@ Providing 2 total locations, including `nifi.registry.extension.dir.1`.
+|Remote Access User|This user name is used to make push requests to the remote repository when `Remote To Push` is enabled, and the remote repository is accessed by HTTP protocol. If SSH is used, user authentication is done with SSH keys.
+|Remote Access Password|Used with `Remote Access User`.
+|====
+
+==== Initialize Git directory
+
+In order to use GitFlowPersistenceRepository, you need to prepare a Git directory on the local file system. You can do so by initializing a directory with `git init` command, or clone an existing Git project from a remote Git repository by `git clone` command.
+
+- Git init command
+https://git-scm.com/docs/git-init
+- Git clone command
+https://git-scm.com/docs/git-clone
+
+
+==== Git user configuration
+
+Git distinguishes a user by its username and email address. This persistence provider uses NiFi Registry username when it creates Git commits. However since NiFi Registry users do not provide email address, preconfigured Git user email address is used.
+
+You can configure Git user name and email address by `git config` command.
+
+- Git config command
+https://git-scm.com/docs/git-config
+
+
+==== Git user authentication
+
+By default, this persistence repository only create commits to local repository. No user authentication is needed to do so. However, if 'Commit To Push' is enabled, user authentication to the remote Git repository is required.
+
+If the remote repository is accessed by HTTP, then username and password for authentication can be configured in the providers XML configuration file.
+
+When SSH is used, SSH keys are used to identify a Git user. In order to pick the right key to a remote server, the SSH configuration file `${USER_HOME}/.ssh/config` is used. The SSH configuration file can contain multiple `Host` entries to specify a key file to login to a remote Git server. The `Host` must much with the target remote Git server hostname.
+
+.example SSH config file
+....
+Host git.example.com
+  HostName git.example.com
+  IdentityFile ~/.ssh/id_rsa
+
+Host github.com
+  HostName github.com
+  IdentityFile ~/.ssh/key-for-github
+
+Host bitbucket.org
+  HostName bitbucket.org
+  IdentityFile ~/.ssh/key-for-bitbucket
+....
+
+=== Data model version of serialized Flow snapshots
+
+Serialized Flow snapshots saved by these persistence providers have versions, so that the data format and schema can evolve over time. Data model version update is done automatically by NiFi Registry when it reads and stores each Flow content.
+
+Here is the data model version histories:
+
+|====
+|*Data model version*|*Since NiFi Registry*|*Description*
+|2|0.2|JSON formatted text file. The root object contains header and Flow content object.
+|1|0.1|Binary format having header bytes at the beginning followed by Flow content represented as XML.
+|====
+
+=== Migrating stored files between different Persistence Provider
```

Comment:
I think instead of providing a tool we can just offer instructions for how to reset your registry to use the git provider, something like:
```
- Stop version control on all PGs in NiFi
- Stop registry
- Move the H2 DB and file-based flow dir somewhere for back up
- Configure git provider in providers.xml
- Start registry
- Recreate any buckets
- Start version control on all PGs again
```
This way the CLI doesn't need to depend on registry framework code since it is more of a client.

What do you think?

Reply:
I agree it would be best to avoid a dependency on nifi-registry-framework for CLI if possible. @bbende, this approach you documented works, but loses the versioned PG history. Not sure how important that is, just mentioning it. 

I think the right analogy for this type of operation is a relational database migration. Most databases support this by giving you some way to export schema and data to a file in a portable SQL syntax that can then be imported into any compatible instance. This feature is useful for snapshotting, creating data backups, or migrating from one DB server to another. If we could support a similar feature in NiFi Registry, as export and import REST API endpoints, and corresponding CLI commands that would call the endpoints and write/read files, that would be a very powerful and flexible capability, both for backing up a NiFi Registry and also changing persistence providers.

This is probably an entirely new feature, so perhaps we could go with @bbende's documentation approach as a stop gap and open a JIRA for a more full-features export/import capability with no data loss.

Reply:
Yea I agree that in the future we could probably offer a bulk import/export via the CLI which only needed to use the REST API. We already have a JIRA for some import/export end-points (https://issues.apache.org/jira/browse/NIFIREG-148), so it might just be some CLI commands to make use of those once they exist.

For this first release since we only have two providers, and since registry is very new, I think it is acceptable to say that if you want to use the git provider then you can start over with it.